tfjs-backend-webgpu/src/kernels/BatchMatMul_impl.ts

```diff
@@ -104,6 +104,19 @@ export function batchMatMulImpl({
   const outputShape: [number, number, number] =
       [batchDim, outerShapeA, outerShapeB];
   let matmulProgramType = env().get('WEBGPU_MATMUL_PROGRAM_TYPE') as number;
+  // Experiments show that when the hardware has more execution units, it's
+  // better to go to MatMulSmallOutputSizeProgram path for small sized output.
+  // Currently, we use threshold 16, which means when the parallel workgroups
+  // number is less than 16 by tiling 32x32. In the satisfied GPUs, increasing
+  // workgroups by a smaller tiling size can get better performance.
+  const matMulSmallOutputSizeProgramContidion =
+      backend.adapterInfo.isIntelGen12GPU &&
+          env().get('WEBGPU_HARDWARE_EU_NUM') as number >= 96 ?
+      Math.ceil(outerShapeA / 32) * Math.ceil(outerShapeB / 32) <= 16 :
+      (outerShapeA <= 16 &&
+       (outerShapeB <= 512 || innerShapeB >= 2 * outerShapeB)) ||
+          (outerShapeB <= 16 &&
+           (outerShapeA <= 512 || innerShapeA >= 2 * outerShapeA));
   if (matmulProgramType < 0) {
```

Comment:
1. Should the defination of `matMulSmallOutputSizeProgramContidion` be in `if (matmulProgramType < 0) {...}` block?
2. Is it better to unify all conditions? Also define `matMulReduceProgramCondition`, `matMulSplitKProgramCondition`,...
3. Use `bool useMatMulSmallOutputSizeProgram = ...`?
All these are only suggestions,  must NOT be addressed.

Reply:
Done.